src/events/eventman.cpp

```diff
@@ -39,6 +75,78 @@ void EventManager::injectKeyboardInput(Events::Key key, Events::KeyState state)
 	}
 }
 
+void EventManager::injectMouse2DAxisInput(Events::Mouse2DAxis axis, glm::vec2 position, glm::vec2 delta) {
+	const auto actions = _mouse2DAxisBindings.equal_range(axis);
+
+	AxisEvent<glm::vec2> axisEvent{position, delta};
+	Event event;
+	event.data = axisEvent;
+
+	for (auto actionIter = actions.first; actionIter != actions.second; ++actionIter) {
+		const auto action = actionIter->second;
+
+		const auto callback = _actionCallbacks[action];
+		event.action = action;
+
+		callback(event);
+	}
+}
+
+void EventManager::injectMouse1DAxisInput(Events::Mouse1DAxis axis, float position, float delta) {
```

Comment:
From what I see in the current state, you are not using the 1D Axis inputs. Do we need them in the future? If not, I would prefer to remove them for simplifying the whole interface

Reply:
Actually, they are used for separating mouse scroll inputs (that are [2D by default](https://www.glfw.org/docs/latest/input_guide.html#scrolling)) into horizontal and vertical components. Here is a snippet for callback assignment from game.cpp:
```cpp
_window->setMouseScrollCallback([&](glm::vec2 absolute, glm::vec2 delta){
		EventMan.injectMouse1DAxisInput(Events::kMouseScrollHorizontal, absolute.x, delta.x);
		EventMan.injectMouse1DAxisInput(Events::kMouseScrollVertical, absolute.y, delta.y);
	});
```